### PR TITLE
Preventing crash due to illegal callback invocation

### PR DESCRIPTION
**Pull request by henninghall**
Solves issue #106

### CODE REVIEW

**Comment by henrikra** [comment]
Nice! Can you reproduce the bug and confirm that this is working? :) Also should the code do something in catch?

**Comment by henninghall** [comment]
I have failed to reproduced it but I haven't had any crashes since 5 months ago when I first implemented it. I know the catch is far from ideal, it is also why I have been waiting and not created an PR until now. Please feel free to improve it if you have an idea :)

**Comment by henrikra** [comment]
@zmxv @benvium @ndbroadbent @TheGhoul21  Can you guys check if we could merge this? If @henninghall has succeeded to get no crashes with this fix for 5 months I think it is worth checking :)

File: android/src/main/java/com/zmxv/RNSound/RNSoundModule.java [hunk 1]
```diff
@@ -46,7 +46,7 @@ public void prepare(final String fileName, final Integer key, final ReadableMap
     }
 
     final RNSoundModule module = this;
-    
+
```
Comment (henrikra) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: Do we need this change? :D

Comment (henninghall) [inline android/src/main/java/com/zmxv/RNSound/RNSoundModule.java]: I can't really confirm that it works without it ;)

**Comment by TheGhoul21** [comment]
The catch() block should IMHO call the error callback in the JS to make it more robust

What do you guys think? Except for that, everything else looks great to me

**Comment by benvium** [comment]
@TheGhoul21 It's possible (perhaps) that this may itself cause an error if the callback has already been invoked..

**Comment by henrikra** [comment]
Should we do release out of this? Since many people have this crash

**Comment by benvium** [comment]
Yes, I'll do one and push to npm.

**Comment by benvium** [comment]
.. but first I'm going to update react-native-sound-demo and make sure everything is working as expected

**Pull request merged**